Program.cs

```diff
@@ -1,4 +1,128 @@
-﻿using System;
-Console.WriteLine ("Hello, World!");
+﻿// --------------------------------------------------------------------------------------------
+// Training ~ A training program for interns at Metamation, Batch - July 2023
+// Copyright (c) Metamation India.                                              
+// ------------------------------------------------------------------------
+// Program.cs
+// MYLIST<T> 
+// Implement a custom MyList<T> class using arrays as the underlying data structure.
+// --------------------------------------------------------------------------------------------
+using System;
+// Test cases
+MyList<int> list = new ();
+list.Add (1);
+list.Add (2);
+list.Add (3);
+list.Add (4);
+list.Add (5);
+list.Add (6);
+list.Add (7);
+list.Display ();
+list.Remove (7);
+list.Clear ();
+list.Display ();
+list.Insert (0, 1);
+list.Insert (1, 2);
+list.Insert (2, 3);
+list.Insert (3, 4);
+list.Insert (4, 5);
+list.Insert (5, 6);
+list.Display ();
+list.RemoveAt (0);
+list.RemoveAt (1);
+list.Display ();
 
-   
\ No newline at end of file
+/// <summary>Create a class</summary>
+/// <typeparam name="T">Datatype of the list</typeparam>
+public class MyList<T> {
+   private T[] mArray;
+   private int mCount;

```

Comment:
Move the declarations to the last. Enclose them within `#region Private data -----`. Similarly, have separate regions for Constructor, Properties and Implementation. For an example, look at ModeSelector.cs file in Spark23> General>Files section